x-pack/plugins/ml/public/application/services/ml_api_service/index.ts

```diff
@@ -640,6 +660,33 @@ export const ml = {
     });
   },
 
+  getModelSnapshots(jobId: string, snapshotId?: string) {
+    return http<GetModelSnapshotsResponse>({
+      path: `${basePath()}/anomaly_detectors/${jobId}/model_snapshots${
+        snapshotId !== undefined ? `/${snapshotId}` : ''
+      }`,
+    });
+  },
+
+  updateModelSnapshot(
+    jobId: string,
+    snapshotId: string,
+    body: { description?: string; retain?: boolean }
+  ) {
+    return http<any>({
+      path: `${basePath()}/anomaly_detectors/${jobId}/model_snapshots/${snapshotId}/_update`,
+      method: 'POST',
+      body: JSON.stringify(body),
+    });
+  },
+
+  deleteModelSnapshot(jobId: string, snapshotId: string) {
+    return http<any>({
+      path: `${basePath()}/anomaly_detectors/${jobId}/model_snapshots/${snapshotId}`,
+      method: 'DELETE',
+    });
+  },
```

Comment:
Could you please create a `SnapshotsService` with `HttpService` dependency instead?

Reply:
i agree that `HttpService` should be injected here, but i think making these changes is outside of the scope of this PR and should be done in a follow up that changes this entire file.